docs/writing-rules/private-rules.md

```diff
@@ -13,40 +13,46 @@ As we continue to develop and refine this feature, we welcome and appreciate all
 
 ## Getting started
 
-***DEPLOYMENT_ID*** can be found in Semgrep App [here](https://semgrep.dev/manage/settings)
+Run 
 
-***SEMGREP_TOKEN*** can be generated in Semgrep App [here](https://semgrep.dev/manage/settings/tokens)
-
-### Dockerhub
-In a directory with the rule file RULE_YAML you want to upload run:
 ```
-docker pull returntocorp/semgrep-upload:latest
-docker run -v $(pwd):/rules -e SEMGREP_UPLOAD_DEPLOYMENT=DEPLOYMENT_ID -e SEMGREP_TOKEN=SOME_TOKEN returntocorp/semgrep-upload:latest /rules/RULE_YAML
+semgrep login
+semgrep publish myrules/
 ```
 
-### Using different registry_url
-
-You can change the registry URL used by setting the SEMGREP_REGISTRY_BASE_URL env variable
+Test cases will be included as well (see [testing Semgrep rules](../writing-rules/testing-rules)).
 
-### Local Dev
+You can also publish rules as unlisted with the same command:
 
 ```
-make setup
-pipenv install
-pipenv run python upload_private_rules.py rule.yaml --deployment_id DEPLOYMENT_ID --token SOME_TOKEN
+semgrep publish --visiblity=unlisted myrules/
 ```
 
-### Docker Dev
-
-```
-make build
-docker run -v $(pwd):/src -e SEMGREP_UPLOAD_DEPLOYMENT=DEPLOYMENT_ID -e SEMGREP_TOKEN=SOME_TOKEN returntocorp/semgrep-upload /src/RULE_YAML
-```
+## Automatically Publishing Rules
 
-### Deploy
+Here is a sample Github Actions workflow to automatically publish rules from a private Git repo after they are merged:
```

Comment:
```suggestion
This section provides a sample of GitHub Actions workflow to automatically publish rules from a private Git repository after they are merged:
```
`after they are merged` - maybe after GitHub Actions are merged?

GitHub, not Github. Repository, not repo.